corehq/form_processor/tests/utils.py

```diff
@@ -172,8 +172,36 @@ def _delete_all(db, viewname, **view_kwargs):
 )
 
 
+def partitioned(cls):
+    """
+    Marks a test to be run with the partitioned database settings in
+    addition to the non-partitioned database settings.
+    """
+    return attr(sql_backend=True)(cls)
+
+
+def only_run_with_non_partitioned_database(cls):
+    """
+    Only runs the test with the non-partitioned database settings.
+    """
+    if settings.USE_PARTITIONED_DATABASE:
+        return nottest(cls)
+
+    return cls
+
+
+def only_run_with_partitioned_database(cls):
```

Comment:
yeah so there are certain tests that i only want to run with a sharded db, and certain ones i only want to run on the monolith db. if i just mark a test with the sql_backend=True attribute, docker picks it up for testing in both kinds of dbs, so in order to only run it for only one type of db we need these extra decorators.

yeah i suppose i could have just used `@skipUnless(settings.USE_PARTITIONED_DATABASE)` as the tests you referenced do, but this moves that logic into a single decorator and makes it a little cleaner. i don't feel too strongly about it though, i can remove this if you prefer i just use `skipUnless`